File: fitness-service/src/main/java/com/frex/fitness/service/FreakInfoService.java
package com.frex.fitness.service;

import com.frex.fitness.exception.FreakInfoAlreadyExistsException;
import com.frex.fitness.exception.FreakInfoDoesNotExistException;
import com.frex.fitness.model.FreakInfo;

public interface FreakInfoService {
    //create
    public boolean saveFreakInfo(FreakInfo freakInfo) throws FreakInfoAlreadyExistsException;

    //retrieve
    public FreakInfo retrieveFreakInfo(String userId, String activityType) throws FreakInfoDoesNotExistException;

    //update
    public boolean updateFreakInfo(FreakInfo freakInfo) throws FreakInfoDoesNotExistException;

    //delete
    public boolean deleteFreakInfo(String userId, String activityType);

    //check
    public boolean checkIfExists(String userId, String activityType);
}

File: frex-ui/src/app/services/fitness/freak-info.service.ts
import {Injectable} from '@angular/core';
import {HttpClient, HttpHeaders} from '@angular/common/http';
import {environment} from '../../../environments/environment';
import {FreakInfo} from '../../util/fitness-activities';

@Injectable({
  providedIn: 'root'
})
export class FreakInfoService {
  private url = environment.fitnessServiceURL + 'freakInfo/';

  constructor(private httpClient: HttpClient) {
  }

  async saveFreakInfo(freakInfo: FreakInfo) {
    return await this.httpClient.post(this.url + 'save', freakInfo).toPromise().then(
      (result) => {
        return result;
      });
  }

  async retrieveFreakInfo(userId: string, activityType: string) {
    return await this.httpClient.get(this.url + 'retrieve', {
      headers: new HttpHeaders().set('userId', userId).append('activityType', activityType)
    }).toPromise().then(
      (value: any) => {
        return value;
      }
    );
  }

  updateFreakInfo(freakInfo: FreakInfo) {
    return this.httpClient.put(this.url + 'update', freakInfo);
  }

  async checkIfExists(userId: string, activityType: string) {
    return await this.httpClient.get(this.url + 'check', {
      headers: new HttpHeaders().set('userId', userId).append('activityType', activityType)
    }).toPromise().then(
      result => {
        return result;
      }
    );
  }
}
